benchmarks/scripts/utils.py

```diff
@@ -198,3 +202,38 @@ def parse_commit_date(value):
         raise ValueError(msg)
 
     return commit_date
+
+
+def parse_capture_time(value):
+    # type: (str) -> float
+    logger.info(value)
+    m = re.match(r"(\d+)([smhd])", value)
+    if not m:
+        raise ValueError(
+            "Got invalid run time: %s, Run time must be in the following format: %s"
+            % (value, "<number><s|m|h|d>, example: 1h - 1 hour")
+        )
+
+    number, time_unit = m.groups()
+    if time_unit == "s":
+        time_unit = "seconds"
+    elif time_unit == "m":
+        time_unit = "minutes"
+    elif time_unit == "h":
+        time_unit = "hours"
+    elif time_unit == "d":
+        time_unit = "days"
+    return timedelta(**{time_unit: int(number)}).total_seconds()
+
+
+def wait_for_agent_start_and_get_pid(pid_file_path):
+    # type: (str) -> int
+    """IF file 'pid_file_path' exists, we assume that the agent is started. Returns 'pid' of the agent process."""
+    attempts = 0
+    while not os.path.exists(pid_file_path):
+        if attempts > 10:
+            raise RuntimeError("Agent pid file did not appear.")
```

Comment:
Perhaps we should also try to cat and include the last 30 lines of the agent.log if available here?

This should help with troubleshooting if there is a typo in a config or similar.

(That's not needed for Circle CI because there we preserve log files as build artifacts, but in may help In Kubernetes and other scenarios)